test/unit/gateways/redsys_sha256_test.rb

```diff
@@ -123,14 +124,49 @@ def test_successful_authorize_with_3ds
     assert_equal response.authorization, '156270437866||'
   end
 
+  def test_successful_purchase_with_3ds
+    @gateway.expects(:ssl_post).returns(successful_authorize_with_3ds_response)
+    response = @gateway.purchase(100, credit_card, { execute_threed: true, order_id: '156270437866' })
+    assert response.test?
+    assert response.params['ds_emv3ds']
+    assert_equal response.message, 'CardConfiguration'
+    assert_equal response.authorization, '156270437866||'
+  end
+
+  def test_successful_purchase_with_3ds2
+    @gateway.expects(:ssl_post).returns(successful_authorize_with_3ds_response)
+    response = @gateway.purchase(100, @threeds2_credit_card, { execute_threed: true, order_id: '156270437866' })
+    assert response.test?
+    assert response.params['ds_emv3ds']
+    assert_equal response.message, 'CardConfiguration'
+    assert_equal response.authorization, '156270437866||'
+  end
+
+  def test_successful_purchase_with_3ds2_and_mit_exemption
```

Comment:
Great new test coverage on the exceptions! Do you think its work adding a test here as well for `DS_MERCHANT_DIRECTPAYMENT`?

Reply:
Yes, but a correction Tatsiana and I made makes it available only on `trataPeticion` and I'm trying to figure out the best way to trigger/capture that.